projects/aws/image-builder/builder/builder.go

```diff
@@ -41,24 +42,43 @@ func (b *BuildOptions) BuildImage() {
 		log.Println("Using repo checked out from code commit")
 	}
 
+	supportedReleaseBranches := GetSupportedReleaseBranches()
+	if !SliceContains(supportedReleaseBranches, b.ReleaseChannel) {
+		cleanup(buildToolingRepoPath)
+		log.Fatalf("release-channel should be one of %v", supportedReleaseBranches)
+	}
+
 	imageBuilderProjectPath := filepath.Join(buildToolingRepoPath, "projects/kubernetes-sigs/image-builder")
 	upstreamImageBuilderProjectPath := filepath.Join(imageBuilderProjectPath, "image-builder/images/capi")
 	var outputArtifactPath string
 	var outputImageGlob []string
+	commandEnvVars := []string{fmt.Sprintf("RELEASE_BRANCH=%s", b.ReleaseChannel)}
 
 	log.Printf("Initiating Image Build\n Image OS: %s\n Hypervisor: %s\n", b.Os, b.Hypervisor)
 	if b.Hypervisor == VSphere {
 		// Read and set the vsphere connection data
-		vsphereConfigData, err := os.ReadFile(b.VsphereConfig)
+		vsphereConfigData, err := json.Marshal(b.VsphereConfig)
 		if err != nil {
-			log.Fatalf("Error reading the vsphere config file")
+			log.Fatalf("Error marshalling vsphere config data")
 		}
 		err = ioutil.WriteFile(filepath.Join(imageBuilderProjectPath, "packer/ova/vsphere.json"), vsphereConfigData, 0644)
 		if err != nil {
 			log.Fatalf("Error writing vsphere config file to packer")
 		}
-		buildCommand := fmt.Sprintf("make -C %s local-build-ova-ubuntu-2004", imageBuilderProjectPath)
-		err = executeMakeBuildCommand(buildCommand, b.ReleaseChannel)
+
+		var buildCommand string
+		switch b.Os {
+		case Ubuntu:
+			buildCommand = fmt.Sprintf("make -C %s local-build-ova-ubuntu-2004", imageBuilderProjectPath)
+		case RedHat:
+			buildCommand = fmt.Sprintf("make -C %s local-build-ova-rhel-7", imageBuilderProjectPath)
+			commandEnvVars = append(commandEnvVars,
+				fmt.Sprintf("RHSM_USERNAME=%s", b.VsphereConfig.RhelUsername),
```

Comment:
Ah I see we pass it here to the image-builder target. What does image builder do with it?

Reply:
These values need to be passed in and set as the values for RHSM_USER and RHSM_PASS env vars for the Ansible step of RedHat subscription - https://github.com/kubernetes-sigs/image-builder/blob/master/images/capi/ansible/roles/setup/tasks/redhat.yml#L15-L21

Reply:
It's required to register with redhat subscription manager to update the image.